include/bluetooth/gatt.h

```diff
@@ -280,7 +286,29 @@ struct bt_gatt_cpf {
  *
  * @return 0 in case of success or negative value in case of error.
  */
-int bt_gatt_register(struct bt_gatt_attr *attrs, size_t count);
+#define bt_gatt_register(_attrs, _count)				\
+	bt_gatt_register_service((&(struct bt_gatt_svc) { .attrs = _attrs, \
+				.attr_count = _count, }))
```

Comment:
I think the only reasonable thing is to require apps to actually create their own static bt_gatt_svc variables.